src/data_structure/segment_tree.rs

```diff
@@ -214,9 +251,13 @@ mod test {
                         }
 
                         assert_eq!(seg.query(i1..i2, j1..j2), Some(minimum));
+                        assert_eq!(seg.query(i1..=(i2-1), j1..j2), Some(minimum));
                     }
                 }
             }
         }
+
+        assert_eq!(seg.query(0..N, ..), seg.query(.., 0..N));
+        assert_ne!(seg.query(0..N, ..), seg.query(0..=N, ..));
```

Comment:
`assert_eq!`?

Reply:
I think in both cases the query is effectively `.query(0..N, 0..N)`, since the `..` should expand to everything, which'll then be reduced to the size of the tree, so they should `assert_eq`

Reply:
Sorry for the confusion. I just wanted to suggest replacing `assert_ne` with `assert_eq` in the second line. I guess it's a typo.

Reply:
Oh I see, didn't notice that slipped by. I should have tested with `cargo test data_structure::segment_tree` otherwise that test didn't even run